test/main.js

```diff
@@ -97,6 +97,29 @@ describe('gulp-stylus', function() {
     stream.end();
   });
 
+  it ('should support defining variables in .styl files using gulp-data (data object attached to file)', function(done) {
+    var stream = stylus();
+    var fakeFile = new gutil.File({
+      base: 'test/fixtures',
+      cwd: 'test/',
+      path: 'test/fixtures/define.styl',
+      contents: fs.readFileSync('test/fixtures/define.styl')
+    });
+
+    fakeFile.data = {'white': '#fff'};
+
+    stream.on('data', function(newFile) {
+      should.exist(newFile);
+      should.exist(newFile.contents);
+
```

Comment:
should.exist(newFile.data) 
should.exist(newFile.data.white) 
should(newFile.data.white).equal('#fff')